src/3d/ocaml/Batch.ml

```diff
@@ -0,0 +1,402 @@
+let code_of_exit = function
+  | Unix.WEXITED i
+  | Unix.WSIGNALED i
+  | Unix.WSTOPPED i
+    -> i
+
+let filename_concat = Filename.concat
+
+let run_cmd_gen reconcat prog args =
+  let cmd = String.concat " " (prog :: args) in
+  print_endline (Printf.sprintf "Running: %s" cmd);
+  let args' = Array.of_list (if reconcat then prog :: args else args) in (* FIXME: WHY WHY WHY do I need to recons the prog in front of the args? *)
+  let pid = Unix.create_process prog args' Unix.stdin Unix.stdout Unix.stderr in
+  let (_, res) = Unix.waitpid [] pid in
+  let res = code_of_exit res in
```

Comment:
Thanks a lot for your suggestion. In fact, with Process, I claim that I should use `execute` (because I don't need to capture output and so I may not want to have output retained in memory), but it is not exposed to the `.mli`.